cpp/core/shuffle/LocalPartitionWriter.cc

```diff
@@ -351,10 +328,6 @@ arrow::Status PreferCachePartitionWriter::stop() {
     auto cachedPayloadSize = shuffleWriter_->partitionCachedRecordbatchSize()[pid];
     if (cachedPayloadSize > 0) {
       if (firstWrite) {
```

Comment:
same as above.

Reply:
Need it, if not first write, will writeEOS.

Reply:
Oh, I was meaning this if-block seems always toggle ```firstWrite``` to ```false```. We can simplify it to ```firstWrite = false;```.